anx/provider/provider_test.go

```diff
@@ -46,6 +48,21 @@ func TestNewProvider(t *testing.T) {
 	require.IsType(t, instanceManager{}, instancesV2)
 	manager := instancesV2.(instanceManager)
 	require.Equal(t, provider, manager.Provider)
+
+	t.Run("with incomplete initialization", func(t *testing.T) {
+		p := &anxProvider{
+			providerMetrics: metrics.NewProviderMetrics("anexia", Version),
```

Comment:
please call `setupProviderMetrics` instead of building it yourself

Reply:
Unfortunately this does not work, as `setupProviderMetrics` calls `legacyregistry.MustRegister(&a.providerMetrics)` which triggers `panic: duplicate metrics collector registration attempted` when called twice. `legacyregistry.Reset()` does not seem to work either.

Reply:
ah, I see - can you add a comment then?